app/components/avatar_component.html.erb

```diff
@@ -1,4 +1,4 @@
 <%= tag.picture do %>
   <%= tag.source media: "(-webkit-min-device-pixel-ratio: 2), (min-resolution: 2dppx)", srcset: avatar_image_2x_url %>
-  <%= image_tag avatar_image_url, alt: "#{name} avatar", data:, class: "#{classes || "h-24 w-24 sm:h-32 sm:w-32 ring-4 ring-white"} object-cover bg-gray-300 rounded-full" %>
+  <%= image_tag avatar_image_url, alt: "#{name} avatar", data:, class: {"#{classes || "h-24 w-24 sm:h-32 sm:w-32 ring-4 ring-white"} object-cover rounded-full": true, "bg-gray-300": !Feature.enabled?(:redesign), "bg-navy": Feature.enabled?(:redesign)} %>
```

Comment:
Are you sure that this isn't working? Here's the generated HTML on this branch for the navbar avatar:
```
<img alt="NilClass's avatar" class="h-10 w-10 rounded-full object-cover rounded-full bg-navy" src="/assets/avatar-cfa52108bb8bf4200e30d39b70ea9715af63299ce52df1e547011a24a2e659f9.png">
```

And here's the generated HTML on the developer page for the redesign branch:
```
<img alt="Developer's avatar" class="h-24 w-24 sm:h-32 sm:w-32 ring-4 ring-white object-cover rounded-full bg-navy" src="/some_activestorage_url">
```

What you were saying about it not working makes sense in context of the code snippet you posted. I'm a bit confused though, because it doesn't match what's in the PR. The class assignment in the PR looks like this:
```
class: {"#{classes || "h-24 w-24 sm:h-32 sm:w-32 ring-4 ring-white"} object-cover rounded-full": true, "bg-gray-300": !Feature.enabled?(:redesign), "bg-navy": Feature.enabled?(:redesign)}
```
So in this case, the default classes are conditionally assigned via string interpolation in the hash. It's definitely a confusing piece of code, but it should hopefully be temporary. If it is working as intended (I'm still not sure, so let me know if I'm missing something here), I don't think it's worth refactoring the component around a temporary feature flag. Regardless, I'll be sure to swap the ordering of the conditionals.

Reply:
Ahhh, gotcha. I was definitely confused by the intent of this code. Which makes me want to refactor it in some way.

For now, let's move the entire `class` option to the `.rb` file of the component. It should be easier to build + read the classes when it's in plain Ruby.